src/runtime/button-api.js

```diff
@@ -98,20 +100,17 @@ export class ButtonApi {
 
   /**
    * @param {!Element} button
-   * @param {!Object|function()} optionsOrCallback
-   * @param {function()=} opt_callback
+   * @param {!../api/subscriptions.ButtonOptions|function()} optionsOrCallback
+   * @param {function():null=} opt_callback
    * @return {!Element}
    */
   attach(button, optionsOrCallback, opt_callback) {
-    const options =
-        typeof optionsOrCallback != 'function' ?
-        optionsOrCallback : null;
-    const callback = /** @type {function()} */ (
-        (typeof optionsOrCallback == 'function' ? optionsOrCallback : null) ||
-            opt_callback);
+    const options = this.getOptions_(optionsOrCallback);
+    const callback = /** @type {function(Event):boolean} */
+        (this.getCallback_(optionsOrCallback, opt_callback));
     let theme = options && options['theme'];
-    if (theme !== 'light' && theme !== 'dark') {
-      theme = 'light';
+    if (theme !== Theme.LIGHT && theme !== Theme.DARK) {
```

Comment:
I think we should move this code to `getOptions_`

Reply:
Re-factored this code to `getOptions_`